core/tests/puppeteer-acceptance-tests/utilities/user/curriculum-admin.ts

```diff
@@ -807,6 +703,274 @@ export class CurriculumAdmin extends BaseUser {
+    await this.page.waitForSelector(skillsTab, {visible: true});
+    await this.clickOn(skillsTab);
+    await this.page.waitForSelector(skillSelector, {visible: true});
+    await this.page.waitForSelector(skillListItemSelector);
+
+    const skills = await this.page.$$(skillListItemSelector);
+    const skillToDelete = skills.find(async skill => {
+      const skillNameElement = await skill.$(skillNameSelector);
+      if (!skillNameElement) {
+        return false;
+      }
+      const name = await (
+        await skillNameElement.getProperty('textContent')
+      ).jsonValue();
+      return name === skillName;
+    });
+
+    if (!skillToDelete) {
+      throw new Error(`Skill ${skillName} not found`);
+    }
+
+    await this.clickElement(skillToDelete, skillListItemOptions);
+    await this.page.waitForFunction(
+      (skill: Element) => !skill.isConnected,
+      {},
+      skillListItemOptions
+    );
+    await this.clickElement(skillToDelete, deleteSkillButton);
+    await this.page.waitForFunction(
+      (button: Element) => !button.isConnected,
+      {},
+      deleteSkillButton
+    );
+    await this.clickElement(this.page, confirmSkillDeletionButton);
+    await this.page.waitForFunction(
+      (button: Element) => !button.isConnected,
+      {},
+      confirmSkillDeletionButton
+    );
+  }
+
+  /**
+   * Function to check if a skill is not present in the Topics and Skills Dashboard.
+   * @param {string} skillName - The name of the skill to check.
+   */
+  async expectSkillNotInTopicsAndSkillsDashboard(
+    skillName: string
+  ): Promise<void> {
+    await this.goto(topicAndSkillsDashboardUrl);
+    const isTextPresent = await this.isTextPresentOnPage(
+      ' No topics or skills have been created yet. '
+    );
+    if (isTextPresent) {
+      showMessage(`The skill "${skillName}" is not present on the Topics and Skills
+      Dashboard as expected.`);
+      return;
+    }
+    await this.clickOn(skillsTab);
+    const isSkillPresent = await this.isTextPresentOnPage(skillName);
+    if (isSkillPresent) {
+      throw new Error(
+        `Skill "${skillName}" was found.
+          It was expected to be absent from Topics and Skills Dashboard.`
+      );
+    }
+    showMessage(
+      `The skill "${skillName}" is not present on the Topics and Skills
+      Dashboard as expected.`
+    );
+  }
+
+  /**
+   * Function to delete all questions in a skill.
+   * @param {string} skillName - The name of the skill to delete questions from.
+   */
+  async removeAllQuestionsFromTheSkill(skillName: string): Promise<void> {
+    await this.openSkillEditor(skillName);
+
+    const isMobileWidth = this.isViewportAtMobileWidth();
+    const skillQuestionTab = isMobileWidth
+      ? mobileSkillQuestionTab
+      : desktopSkillQuestionTab;
+
+    if (isMobileWidth) {
+      await this.clickOn(mobileOptionsSelector);
+      await this.page.screenshot({path: 'debug1.png'});
```

Comment:
Why are you taking screenshots? If you are not using it for debugging locally, why are the screenshots not uploaded in the PR? Ditto for the two screenshots below.

Reply:
I was taking the screenshots to debug locally, have removed the snapped screenshots and screenshot statements in the latest commit as well.